spec/spec_helper.rb

```diff
@@ -51,10 +52,18 @@ def self.run_all(reporter = nil)
   config.warnings = true
   config.raise_on_warning = true
 
+  # Execute a provided block with RSpec global objects (configuration,
+  # world, current example) reset. This is used to test specs with RSpec.
   config.around(:example) do |example|
-    real_world = RSpec.world
-    RSpec.instance_variable_set(:@world, RSpec::Core::World.new)
-    example.run
-    RSpec.instance_variable_set(:@world, real_world)
+    RSpec::Core::Sandbox.sandboxed do |sandbox_config|
+      # If there is an example-within-an-example, we want to make sure the inner
+      # example does not get a reference to the outer example (the real spec) if
+      # it calls something like `pending`.
+      sandbox_config.before(:context) { RSpec.current_example = nil }
+      RSpec::Rails.initialize_configuration(sandbox_config)
+      example.run
+    end
   end
+
+  config.include RSpec::Rails::FeatureCheck, type: :generator
```

Comment:
Can we not just include this where it's used?

Reply:
We can, but `type_metatag` is used in every one of them except for install generator spec.

Reply:
I intended to get rid of this method altogether, replacing all usages to `type: :some_type`, and this `include` as well.
Ok to keep it here and remove in a follow-up?
Or keep `type_metatag` and inline `include`s into generator specs?
Or keep this include as is and leave `type_metatag` alone?
I don't have a preference here, up to you.

Reply:
So you can remove this from here, and add it here:

```
module RSpec
  module Rails
    module Specs
      module Generators
        module Macros
        # spec/support/generators.rb:18
        def self.included(klass)
          klass.extend(Macros)
          klass.include(RSpec::Rails::FeatureCheck)
        end
```

Reply:
Good point. Done.